### PR TITLE
Create README with maintenance instructions

### LINKED ISSUE #58: Create a readme with maintenance instructions for future classes
I can write something up -- we can obviously be in contact (e.g. me bugging erik) with whoever takes over this, but I thought it might be useful to write some stuff down: creating a form for the class, setting it up with the data document, inputting coordinates/logos for new schools, etc.

### LINKED ISSUE #61: Create comment in script.js to explain dataDocuments variable
We need to note that the dataDocuments variable must be updated each year with the new year's data document URL. Put a comment on top of it.

> Hmm... I don't love explicitly stating what code to look for, because it can change without this guide changing with it. Can anyone think of a simpler way to specify this? Maybe we can mark it with an inline comment?

_Originally posted by @ErikBoesen in https://github.com/apc-gmhs/seniormap/pull/59#discussion_r619710550_

### PR DESCRIPTION
**Pull request by JonathanOppenheimer**
Fairly in-depth instructions about site maintenance. Absolutely suggest content changes as you all are more familiar with the project than I am. 

A couple of notes 
- I tried to get the images to appear directly in the readme instead of being links, but it was kinda a pain and if we wanted to do so the easiest way would be to upload them to the repo and go from there -- currently, they're just linked from the G-drive
- Information about updating the form link would need to be changed if we ever change the URL form link to be specific to each year
- I think I covered everything that needs to be done? Might have missed some aspects

### CODE REVIEW

File: README.md
Comment (ErikBoesen) [inline README.md]: * club -> Club
* Rather than explicitly writing the link out, I'd prefer it to be represented as a MarkDown link. Also no trailing slash needed.

File: README.md
Comment (ErikBoesen) [inline README.md]: Perhaps explain how you can get access to this folder. (Even if that's just "ask x for the link")

Might be worth dropping a link here, and people without access can request it through the official Google Drive access request thing

File: README.md
Comment (ErikBoesen) [inline README.md]: Mind just embedding this image directly rather than linking the Google Drive file? It's ok if you need to make a `readme` dir to store images in.

Comment (AndrewLester) [inline README.md]: The Google spreadsheet must be published to the web. Add this as a fourth step

File: README.md
Comment (ErikBoesen) [inline README.md]: Hmm... I don't love explicitly stating what code to look for, because it can change without this guide changing with it. Can anyone think of a simpler way to specify this? Maybe we can mark it with an inline comment?

File: README.md
Comment (AndrewLester) [inline README.md]: I say put periods at the end of these steps.

File: README.md
Comment (ErikBoesen) [inline README.md]: Are these non-breaking spaces intentional? If not, definitely remove them

File: README.md
Comment (AndrewLester) [inline README.md]: Typo (a the)

File: README.md
Comment (ErikBoesen) [inline README.md]: Same here

File: README.md
Comment (ErikBoesen) [inline README.md]: Wait, do we not just show the form link for the year that's being viewed? I thought we did. If not, we definitely should.

File: README.md
Comment (AndrewLester) [inline README.md]: Typo (list of student)

File: README.md
Comment (AndrewLester) [inline README.md]: Python with a capital P

File: README.md
Comment (ErikBoesen) [inline README.md]: Perhaps you should clarify/reword this a bit or add a screenshot, it's not really clear  to me what this means. Also mimicking is spelled wrong.

File: README.md
Comment (AndrewLester) [inline README.md]: Specify the Location Information spreadsheet inside the shared data folder

File: README.md
Comment (AndrewLester) [inline README.md]: Insert step about using the debug functions.

File: README.md
Comment (AndrewLester) [inline README.md]: Talk to class officers for this.

File: README.md
Comment (ErikBoesen) [inline README.md]: student -> students

Questions a reader might have that we should answer: Who is a registrar? How can I get their contact information? Who is the current registrar?

Also the second sentence here is incoherent

**Review by AndrewLester — CHANGES_REQUESTED** [review]
Looks good. Yeah we should probably upload photos to the repo.

File: README.md
Comment (ErikBoesen) [reply README.md]: Also, prefer "they" over "he/she"

File: README.md
Comment (ErikBoesen) [inline README.md]: He/she -> they

File: README.md
Comment (ErikBoesen) [inline README.md]: Are we 100% sure this script always works? Should we give directions explaining how manually renaming the files might work?

File: README.md
Comment (ErikBoesen) [inline README.md]: Reword this and add detail, it's not clear what this means

File: README.md
Comment (ErikBoesen) [inline README.md]: Nit: python -> Python

"Ensure the lack of discrepancies" is a strange way to say this. Perhaps "Ensure there are no incorrect names" or something?

File: README.md
Comment (ErikBoesen) [inline README.md]: I think this list would be slightly more readable if you didn't remove the punctuation at the end of the lines. Especially since there are often multiple complete sentences in a single bullet point, it just looks a little weird to drop the period off the last one.

File: README.md
Comment (ErikBoesen) [inline README.md]: Add periods here too

File: README.md
Comment (AndrewLester) [reply README.md]: Also this should say "ensure the form"

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: If I get around to adding a debug function for the images that don't load correctly, like you suggested, definitely. What else is there to say about the debug function?

**Comment by JonathanOppenheimer** [comment]
> Looks good. Yeah we should probably upload photos to the repo.

I'll upload them right now, but I think I'll have to commit directly to the readme as you just use the links inside of the repository - after pushing the links for the images will change. 
`alt text`

File: README.md
Comment (ErikBoesen) [inline README.md:40]: Could we do this with conditional formatting somehow? What's the utility of this?

Comment (ErikBoesen) [reply README.md:40]: Forgive my lack of context, I haven't actually been performed this process since my own grad year and it seems like a lot has changed.

File: README.md
Comment (ErikBoesen) [inline README.md]: Trim `/view?usp=sharing` off the end of these links.

File: README.md
Comment (ErikBoesen) [inline README.md]: *Institutions

Also trim link

File: README.md
Comment (ErikBoesen) [inline README.md]: Consider adding documentation of what our normal avenues of publicity are

**Review by ErikBoesen — CHANGES_REQUESTED** [review]

File: README.md
Comment (AndrewLester) [reply README.md]: I think we should point out that the script just goes through ever image and applies names to them alphabetically based on their numerical order. Basically whoever runs it needs to be aware they need to set everything up right.

File: README.md
Comment (AndrewLester) [reply README.md]: Basically you can check for colleges with missing logos.

File: README.md
Comment (ErikBoesen) [reply README.md]: @AndrewLester I'm curious—is that ever actually correct? I feel like there would be guaranteed to be SOME person who's missing in the spreadsheet or missing in the photos list and then everything would be off by X.

Comment (AndrewLester) [reply README.md]: Oh there most definitely is. There is a large process of modifying the list of names to fit the existing pictures. It's not as bad as if the pictures weren't alphabetically numbered.

**Comment by AndrewLester** [comment]
I just realized, we are going to need more info to note that all grades must pass on the Google cloud platform project.

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: I like linking it and having people ask for access - good thinking

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: Yeah they were accidental from copying from a format website - I changed to sub-bullets

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: We do not! It's been like this since August haha (https://github.com/apc-gmhs/seniormap/issues/39)

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: I've included a link to the admin staff directory and I'll put the current registrar's name as well. Knowing GM, they'll switch up their website next week, and replace the registrar next month, so this will probably have to be updated in the future.

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: I've added a brief disclaimer about setting it up correctly - Andrew if you think we should do a (lengthier) write-up about how we checked and modified the list against that thumbnail document, I could add it in. It's fairly obvious when it *doesn't* work correctly and the names are off so I'm hoping there will never be a scenario where future seniors upload a list of names with everything off by 4 names.

File: README.md
Comment (TheBlueness) [reply README.md:43]: > I just realized, we are going to need more info to note that all grades must pass on the Google cloud platform project.

Would you mind writing up a little explanation about what this does/why it needs to be done? I'm thinking we could add another subsection with information about "passing on the map" with information like transferring ownership of the senior map, giving access to the Github repo, etc. if you all think that would be valuable.

**Comment by JonathanOppenheimer** [comment]
> I just realized, we are going to need more info to note that all grades must pass on the Google cloud platform project.

Would you mind writing up a little explanation about what this does/why it needs to be done? I'm thinking we could add another subsection with information about "passing on the map" with information like transferring ownership of the senior map gdrive folder, giving access to the Github repo, etc. if you all think that would be valuable.

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: Alright I changed the verbiage a bit -- is that more what you were going for?

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: Yeah embedding the images would look much cleaner and make the whole tutorial easier to understand. The images are being uploaded in this PR and after it gets merged I can then embed all the images directly by doing: 

`alt text`

I could do them beforehand by uploading the Imgur or something but that's gross -- I just don't know what the URL will be for the images

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: Putting some kind of comment flag in scripts.js that we tell them to look for sounds good to me - @AndrewLester thoughts?

File: README.md
Comment (ErikBoesen) [reply README.md]: You don't have to do that. Just do
```
![alt](tutorial_images/tutX.png)
```
It will work without the full link; I've done it before successfully.

Also, I would prefer the directory of images be called `readme`.

Comment (JonathanOppenheimer) [reply README.md]: Oh, nice - always learning new things! I'll make both changes

**Comment by ErikBoesen** [comment]
@TheBlueness  I like that idea. Also, write commits in imperative tense: added/updated/etc -> Add/Update...

**Comment by JonathanOppenheimer** [comment]
> @TheBlueness I like that idea. Also, write commits in imperative tense: added/updated/etc -> Add/Update...

Another thing I can start doing haha - once Andrew shares his thoughts I can start writing up a "passing on the map" section

**Comment by AndrewLester** [comment]
Here are my passing on the map steps:

- Find an experienced leadership member of the APC who is a student in the next graduating class to take over the map.
- Request their personal email and transfer ownership of the Senior Map Google Cloud Project to them.
- Give their student email account ownership of all files (we need a better way to do this, perhaps just give an external Google account ownership permanently, and add people as necessary)?
- Make sure they can access the repo as editor.

**Comment by JonathanOppenheimer** [comment]
> * Give their student email account ownership of all files (we need a better way to do this, perhaps just give an external Google account ownership permanently, and add people as necessary)?

We've always got the APC email which we could make own the folder - I'll put in your steps.

**Comment by AndrewLester** [comment]
Good idea. ~~I'll make the APC Gmail owner of all files~~. Student's can become owner of the Senior map folder. Files are only deleted when the owner account of the file is deleted.

EDIT: Files can't be transferred outside of organization. We'd have to recreate them all inside the APC drive. This is doable, but for another time.

File: README.md
Comment (AndrewLester) [reply README.md]: Sure. There's now a separate issue referencing this. In the step in the README, just put something like "read the scripts.js file for necessary updates that must be made every year).

**Comment by ErikBoesen** [comment]
@AndrewLester Just make sure that the forms/sheets are owned by a student in fccps.org—I think they need to be so that we can limit form submissions to that org.

**Comment by AndrewLester** [comment]
> @AndrewLester Just make sure that the forms/sheets are owned by a student in fccps.org—I think they need to be so that we can limit form submissions to that org.

Hmm okay I forgot about this. The ideal scenario is a shared drive, but I think even administrators can't make those anymore. The only one that exists is the robotics one.

**Review by AndrewLester — DISMISSED** [review]
Looks good to me. I don't know if we are waiting for #39 to get closed before or after merging this. I think we can get that done soon, so maybe we just wait on this PR and once it's complete we can add one last step based on the implementation of the year-based form URL.

File: README.md
Comment (AndrewLester) [reply README.md]: Yeah this is good.

File: README.md
Comment (ErikBoesen) [inline README.md:31]: Damn this is quite a website to recommend

File: js/scripts.js
Comment (ErikBoesen) [inline js/scripts.js]: Nit: space between `//` and `README`

**Review by ErikBoesen — DISMISSED** [review]

**Comment by JonathanOppenheimer** [comment]
> Looks good to me. I don't know if we are waiting for #39 to get closed before or after merging this. I think we can get that done soon, so maybe we just wait on this PR and once it's complete we can add one last step based on the implementation of the year-based form URL.

Sounds good -- we can let this fella sit tight and whenever the form links get changed I can put a step in about how to add the new year to whatever the system ends up being.

**Comment by AndrewLester** [comment]
#39 is now closed, and you can now add the step for updating a year's form URL. Basically you just need to put the url for the form (the submission URL) in the Year Data spreadsheet under the Form URL column in the same row as the current year.

**Review by AndrewLester — DISMISSED** [review]

File: README.md
Comment (ErikBoesen) [inline README.md]: Bottom left?

File: README.md
Comment (ErikBoesen) [inline README.md]: Replace link with just `resources/tools/namePictures.py`. You don't need the whole link when you're reading on GitHub.

File: README.md
Comment (ErikBoesen) [inline README.md]: tut2 comes after tut3 and 4—maybe we should rename these images to describe what's actually in them rather than their order in the document, that way there's no numbers to change if we reorder the README.

File: README.md
Comment (ErikBoesen) [inline README.md]: Might be worthwhile to add a list of previous maintainers and their emails/GH accounts

Comment (JonathanOppenheimer) [reply README.md]: I'll link the Github account - if future maintainers want to have an email available it can always be listed on their GitHub account profile page.

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: definitely a good idea - I didn't do it linearly so it got messed up.

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: I added you too which I hope was okay -- lmk if you want to be removed

File: README.md
Comment (JonathanOppenheimer) [reply README.md]: This was fixed by Andrew in #64

**Review by AndrewLester — DISMISSED** [review]
Make sure to fix conflicts.

**Review by AndrewLester — DISMISSED** [review]

**Review by AndrewLester — DISMISSED** [review]
@ErikBoesen I think this looks good. You can give it another once over if you'd like, but then we can merge it.

**Review by ErikBoesen — CHANGES_REQUESTED** [review]
Can you rename the readme images from camelCase to snake_case?

**Comment by JonathanOppenheimer** [comment]
> Can you rename the readme images from camelCase to snake_case?

yessir

**Review by AndrewLester — APPROVED** [review]

**Review by ErikBoesen — APPROVED** [review]

**Pull request merged**
